fathom/src/surface/pretty.rs

```diff
@@ -293,6 +295,16 @@ impl<'interner, 'arena> Context<'interner, 'arena> {
                 self.text(","),
                 self.text("}"),
             ),
+            Term::BinOp(_, lhs, op, rhs) => self.paren(
+                prec > op.precedence(),
+                self.concat([
+                    self.term_prec(Prec::Atomic, lhs),
+                    self.space(),
+                    self.text(op.as_str()),
+                    self.space(),
+                    self.term_prec(Prec::Atomic, rhs),
+                ]),
```

Comment:
Oh, the precedences of the operators need to match the grammar

Reply:
I pushed some changes for this but I'm not sure if I did it correctly.